File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.py
class Solution(object):
    def checkSubarraySum(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """
        if len(nums) < 2:
            return False
        
        if k == 0:
            for i in range(1, len(nums)):
                if nums[i-1] == number[i] == 0:
                    return True
            return False
        
        hmap = {}
        hmap[0] = True
        pSum = nums[0]
        
        for i in range(1, len(nums)):
            oldSum = pSum
            pSum = (pSum + nums[i]) % k
            
            if pSum in hmap:
                return True
            hmap[oldSum % k] = True
        
        return False
        
        

File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.cpp
class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) {
        
        if (nums.size() < 2)
            return false;
        
        if ( k == 0) {
            for (int  i = 1; i < nums.size(); i++) {
                if ( (nums[i-1] == 0) && (nums[i] == 0))
                    return true;
            }
            
            return false;
        }
        
        unordered_map<int, bool> hmap;
        hmap[0] = true;
        
        int pSum = nums[0];
        for ( int i = 1; i < nums.size(); i++) {
            int oldSum = pSum;
            pSum = (pSum + nums[i]) % k;
            
            if (hmap.find(pSum) != hmap.end())
                return true;
            
            hmap[oldSum % k] = true;
        }
        
        return false;
    }
};